Sources/Sentry/SentrySessionTracker.m

```diff
@@ -111,10 +130,21 @@ - (void)endCachedSession
 }
 
 /**
- * Is only called when an app is receiving events / it is in the foreground.
+ * Is called when an app is receiving events / it is in the foreground and when we receive a
+ * SentryHybridSdkDidBecomeActiveNotification. There is no guarantee if this method is called once
+ * or twice. We need to ensure that we execute it only once.
  */
 - (void)didBecomeActive
 {
+    // We don't know if the hybrid SDKs post the notification from a background thread, so we
+    // synchronize to be safe.
```

Comment:
the sync. is needed, but the comment is off for me, it's not really about a background thread, it's about receiving 2 notifications at the same time and racing each other

Reply:
If we would receive both notifications on the main thread, they can't race. Usually, you receive the `didBecomeActive` notification on the main thread, so we don't have any synchronization right until now. The callback is executed on the thread you post the notification. As I don't know on which thread the hybrid SDKs might post the notification I added the synchronize.

Reply:
I do agree with you, but what I mean is, I'm not entirely sure that the main thread on Android OS is actually the very same main thread on Flutter, as they use the Message channel for these things, still, it'd be considered the main thread.
reading again your comment, it does make sense.